scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -55,6 +55,44 @@
       (catch Exception e
         (log/error e "Error processing status update")))))
 
+(defn task-ents->map-by-task-id
+  [task-ents]
+  (->> task-ents
+       (map (fn [task-ent] [(str (:instance/task-id task-ent)) task-ent]))
+       (into {})))
+
+(defn task-ent->expected-state
+  [task-ent]
+  
+  (case (:instance/status task-ent)
+    :instance.status/unknown {:expected-state :expected/starting}
+    :instance.status/running {:expected-state :expected/running}
+    :instance.status/failed {:expected-state :expected/completed}
+    :instance.status/success {:expected-state :expected/completed}))
+
+(defn determine-expected-state
+  "We need to determine everything we should be tracking when we construct the expected state. We should be tracking all tasks that are in the running state as well
+  as all pods in kubernetes. We're given an already existing list of all running tasks entities (via (->> (cook.tools/get-running-task-ents)."
+  [conn compute-cluster-name running-tasks-ents current-pods-atom]
+  (let [db (d/db conn)
+        all-tasks-ids-in-pods (->> @current-pods-atom keys (into #{}))
+        _ (log/info "All tasks in pods: " all-tasks-ids-in-pods)
+        running-tasks-in-cc-ents (filter
+                                   #(-> % cook.task/task-entity->compute-cluster-name (= compute-cluster-name))
+                                   running-tasks-ents)
+        _ (log/info "Running tasks in compute cluster in datomic: " running-tasks-in-cc-ents)
+        cc-running-tasks-map (task-ents->map-by-task-id running-tasks-in-cc-ents)
+        cc-running-tasks-ids (->> cc-running-tasks-map keys (into #{}))
+        extra-tasks-map (->> (set/difference all-tasks-ids-in-pods cc-running-tasks-ids)
+                             (map (fn [task-id] [task-id (cook.tools/retrieve-instance db task-id)]))
+                             (filter (fn [[_ task-ent]] (some? task-ent)))
+                             (into {}))
+        all-tasks-ents-map (set/union extra-tasks-map cc-running-tasks-map)]
+    (doseq [[k v] all-tasks-ents-map]
+      (log/info "Setting expected state for " k " ---> " (task-ent->expected-state v)))
```

Comment:
This feels like a `debug` log. Can we do one info level log at the end with the count or something like that?

Reply:
Sounds good.